src/mbgl/annotation/annotation_manager.cpp

```diff
@@ -46,6 +58,7 @@ AnnotationID AnnotationManager::addAnnotation(const Annotation& annotation) {
 }
 
 bool AnnotationManager::updateAnnotation(const AnnotationID& id, const Annotation& annotation) {
+    CHECK_ANNOTATIONS_ENABLED_AND_RETURN(true);
```

Comment:
`CHECK_ANNOTATIONS_ENABLED_AND_RETURN(dirty);` ?

Reply:
`dirty` is not assigned if annotations are disabled, so we just always return `true` in that case

Reply:
this is the same as ^^^, may be discarded.